File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/Polimorfismo/Ejer1(py y java)/FiguraGeometrica.java
// a) 
public class FiguraGeometrica {
    protected String nombre;

    public FiguraGeometrica(String nombre){
        this.nombre=nombre;
    }
}

// a) Cuadrado
class Cuadrado extends FiguraGeometrica{
    private double lado;

    public Cuadrado(double lado){
        super("Cuadrado");
        this.lado=lado;
    }

    // a) Á.
    public double area(){
        return lado*lado;
    }

    // b) P.
    public double perimetro(){
        return 4*lado;
    }
}

// a) Triangulo
class Triangulo extends FiguraGeometrica{
    private double base,altura,lado1,lado2,lado3;

    public Triangulo(double base,double altura,double lado1,double lado2,double lado3){
        super("Triangulo");
        this.base=base;
        this.altura=altura;
        this.lado1=lado1;
        this.lado2=lado2;
        this.lado3=lado3;
    }

    // a) Á.
    public double area(){
        return (base*altura)/2;
    }

    // b) P.
    public double perimetro(){
        return lado1+lado2+lado3;
    }
}

// a) Circulo
class Circulo extends FiguraGeometrica{
    private double radio;

    public Circulo(double radio){
        super("Circulo");
        this.radio=radio;
    }

    // a) Á.
    public double area(){
        return Math.PI*radio*radio;
    }

    // b) P.
    public double perimetro(){
        return 2*Math.PI*radio;
    }
}

// a) Rectangulo
class Rectangulo extends FiguraGeometrica{
    private double base,altura;

    public Rectangulo(double base,double altura){
        super("Rectangulo");
        this.base=base;
        this.altura=altura;
    }

    // a) Á.
    public double area(){
        return base*altura;
    }

    // b) P.
    public double perimetro(){
        return 2*(base+altura);
    }
}

// d) 
class FiguraGeometricas{
    public static void main(String[] args){
        Cuadrado cuad=new Cuadrado(5);
        System.out.println("cuadrado - area:"+cuad.area()+" perimmetro:"+cuad.perimetro());

        Triangulo tri=new Triangulo(4,3,4,5,3);
        System.out.println("triangulo - area:"+tri.area()+" perimetro:"+tri.perimetro());

        Circulo circ=new Circulo(7);
        System.out.println("circulo - area:"+Math.round(circ.area()*100.0)/100.0+" perimetro:"+Math.round(circ.perimetro()*100.0)/100.0);

        Rectangulo rect=new Rectangulo(6,3);
        System.out.println("rectángulo - area:"+rect.area()+" perimetro:"+rect.perimetro());
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/Polimorfismo/Ejer1(py y java)/FiguraGeometrica.py
import math

# a)
class FiguraGeometrica:
    def __init__(self,nombre):
        self.nombre=nombre

# a) Cuadrado
class Cuadrado(FiguraGeometrica):
    def __init__(self,lado):
        super().__init__("cuadrado")
        self.lado=lado

    # a) Á.
    def area(self):
        return self.lado*self.lado

    # b) P.
    def perimetro(self):
        return 4*self.lado

# a) Triángulo
class Triangulo(FiguraGeometrica):
    def __init__(self,base,altura,lado1,lado2,lado3):
        super().__init__("triangulo")
        self.base=base
        self.altura=altura
        self.lado1=lado1
        self.lado2=lado2
        self.lado3=lado3

    # a) Á.
    def area(self):
        return (self.base*self.altura)/2

    # b) P.
    def perimetro(self):
        return self.lado1+self.lado2+self.lado3

# a) Círculo
class Circulo(FiguraGeometrica):
    def __init__(self,radio):
        super().__init__("circulo")
        self.radio=radio

    # a) Á.
    def area(self):
        return math.pi*self.radio**2

    # b) P.
    def perimetro(self):
        return 2*math.pi*self.radio

# a) Rectángulo
class Rectangulo(FiguraGeometrica):
    def __init__(self,base,altura):
        super().__init__("rectangulo")
        self.base=base
        self.altura=altura

    # a) Á.
    def area(self):
        return self.base*self.altura

    # b) P.
    def perimetro(self):
        return 2*(self.base+self.altura)

#d)
cuad=Cuadrado(5)
print("cuadrado-area:",cuad.area(),"perimetro:",cuad.perimetro())

tri=Triangulo(4,3,4,5,3)
print("triangulo-area:",tri.area(),"perimetro:",tri.perimetro())

circ=Circulo(7)
print("circulo-area:",round(circ.area(),2),"perimetro:",round(circ.perimetro(),2))

rect=Rectangulo(6,3)
print("rectángulo-area:",rect.area(),"perimetro:",rect.perimetro())
